<launch>
   
    <node type="conveyor_detector_node" pkg="visy_detector_pkg" name="conveyor_detector_node" output="screen">
      <param name="image_transport" value="compressed" />
    </node>

    <node type="metal_chip_detector_node" pkg="visy_detector_pkg" name="metal_chip_detector_node" output="screen">
      <param name="image_transport" value="compressed" />
    </node>

    <arg name="enable_raw" default="false"/>
    <arg name="enable_imv" default="false"/>
    <arg name="camera_id" default="0"/>
    <arg name="camera_frame_id" default="raspicam"/>
    <arg name="camera_name" default="camerav2_410x308"/>

    <node type="raspicam_node" pkg="raspicam_node" name="raspicam_node" output="screen">
      <param name="private_topics" value="true"/>

      <param name="camera_frame_id" value="$(arg camera_frame_id)"/>
      <param name="enable_raw" value="$(arg enable_raw)"/>
      <param name="enable_imv" value="$(arg enable_imv)"/>
      <param name="camera_id" value="$(arg camera_id)"/>

      <param name="camera_info_url" value="package://raspicam_node/camera_info/camerav2_410x308.yaml"/>
      <param name="camera_name" value="$(arg camera_name)"/>
      <param name="width" value="410"/>
      <param name="height" value="308"/>

      <param name="framerate" value="30"/>
      <param name="exposure_mode" value="antishake"/>
      <param name="shutter_speed" value="0"/>
    </node>

    <node name="$(anon dynparam)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters raspicam_node">
      <param name="contrast" type="int" value="0" />
      <param name="sharpness" type="int" value="0" />
      <param name="brightness" type="int" value="50" />
      <param name="saturation" type="int" value="0" />
      <param name="ISO" type="int" value="700" />
      <param name="exposure_compensation" type="int" value="0" />
    </node>

</launch>
